platform/core/src/classes/metadata/StudyMetadata.js

```diff
@@ -393,6 +386,95 @@ export class StudyMetadata extends Metadata {
     }
   }
 
+  /**
+   * Insert the displaySet so that the list has an increasing SeriesNumber,
+   * with the most recent series first for displaySets with the same SeriesNumber.
+   *
+   * If the displaySet is low priority, the same logic is applied, but is sorted within a sub list
+   * At the end of the list, where all low priority data is found.
+   */
+  _insertDisplaySet(displaySet) {
+    const { SeriesNumber } = displaySet;
+    const displaySets = this._displaySets;
+    let insertIndex = displaySets.length;
+    let firstIndexWithSameSeriesNumber;
+
+    // If low priority, start search from next low priority.
+    if (isLowPriorityModality(displaySet.Modality)) {
+      let startingIndex;
+
+      // Find where the first low priority displaySet is.
+      for (let i = 0; i < displaySets.length; i++) {
```

Comment:
optimization, so not blocking: we can cache this?

Reply:
Or we can maintain these as separate lists internally, and then provide "displaysets" by joining the two

Reply:
Good points, we can keep further optimisations in mind if speed becomes an issue (maybe with multiple hundred SRs?), but I think we've already spent enough IDC time sorting the list.